superset-embedded-sdk/src/index.ts

```diff
@@ -116,6 +117,8 @@ export async function embedDashboard({
         + filterConfigKeys
           .map(key => DASHBOARD_UI_FILTER_CONFIG_URL_PARAM_KEY[key] + '=' + filterConfig[key]).join('&')
         : ""
+      let extraUrlParams = new URLSearchParams(dashboardUiConfig?.urlParams || window.location.search).toString()
```

Comment:
Introducing this now could be seen as a breaking change and has security implications, as it will start sending through all url params. How do you feel about the following?

1. Adding the possibility of providing static mappings (the `dashboardUiConfig?.urlParams` part in this PR).
2. Adding the possibility to let through all params (the `window.location.search` part in this PR).
3. Adding the possibility to allowlist specific URL params (not made possible here)

I think we should not assume 2 unless it's been explicitly allowed. WDYT?

Reply:
Hi @villebro, I totally agree with your concern.

FYI, I appended the `window.location.search` part to increase intuitive. because most of the users expects that all url params on main page will applied to embedded dashboard.

I think we should replace `window.location.search` to `{}`. and need to provides more detailed instructions about it.

So we can cover 'General Case' and 'Special Case' I wrote above using `Adding the possibility of providing static mappings`.

Thanks & Regards.

Reply:
Hi @villebro, I've been thinking...

How about **passing query parameters which has prefix(like '`superset__`')** for Superset Embedded Dashboard.

It can cover problems that we are aware of.

I think it's better. WDYT?

Reply:
@jileeon, IIUC it would still make it possible to pass arbitrary query params from the embedded dashboard to Superset. I think it would be nice to have a scheme for allowlisting or mapping specific query params, so that admins have control over which params are allowed.

How would you feel about the following: we add a parameter `allow_embedded` to the jinja `url_param` function signature that defaults to `False`, so that the person responsible for the chart/dataset that contains jinja has control of either allowing or disallowing embedded query params through? Example:

```python
{{
    url_param("gender", allow_embedded=True)  # Would allow embedded query params through
}}
```

vs.

```python
{{
    url_param("gender")  # Would default to not allowing them through
}}
```

This would ensure we don't introduce a breaking security change, and would open up support for allowing arbitrary url params.

Reply:
@villebro, Yes it is right If an embedded dashboard can be accessed by public internet(not a restricted network).
So, we should consider the possibility of passing arbitrary query params(like you said above). and your proposal about securing Embedded Dashboards using `allow_embedded` is necessary.

So, Do you agree with terms below?
1. Any query params can be passed to an embedded dashboard - It means that this merge request is quite reasonable for this term.
2. Provides the feature you proposed above, So it can secure Embedded Dashboards from passing arbitrary query params. - One more thing, I want a feature flag to enable it globally because I'm using Superset for an internal service.

Thank you for your deep consideration! :)